### PR TITLE
docs[community]: Fix raw string in docstring

### LINKED ISSUE #26212: SyntaxWarning: invalid escape sequence '\s'
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```shell
mypy .
```

### Error Message and Stack Trace (if applicable)

```shell
./python3.12/site-packages/langchain_community/document_loaders/recursive_url_loader.py:54: SyntaxWarning: invalid escape sequence '\s'
  """Recursively load all child links from a root URL.
```

image

### Description

When I run mypy for the first time (if `.mypy_cache` exists, it must be deleted), I always get this warning.
I searched for this error message, and it might only appear in 3.12.

Do we need to add `r""` to that docstring? Will there be any side effects?

### System Info

langchain==0.2.16
langchain-community==0.2.16
langchain-core==0.2.38

platform==linux
python-version==3.12.5

### PR DESCRIPTION
**Pull request by tibor-reiss**
Fixes #26212: replaced the raw string with backslashes. Alternative: raw-stringif the full docstring.

### CODE REVIEW

**Comment by efriis** [comment]
will comment in issue. These strings are identical, so won't have any impact on folks, but sounds like they might be running a very old, or a newer+buggy version of mypy

File: libs/community/langchain_community/document_loaders/recursive_url_loader.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def simple_metadata_extractor(
                 "https://docs.python.org/3.9/",
                 prevent_outside=True,
                 base_url="https://docs.python.org",
-                link_regex=r'<a\s+(?:[^>]*?\s+)?href="([^"]*(?=index)[^"]*)"',
+                link_regex='<a\\s+(?:[^>]*?\\s+)?href="([^"]*(?=index)[^"]*)"',
```
Comment (efriis) [inline libs/community/langchain_community/document_loaders/recursive_url_loader.py:230]:
```suggestion
                link_regex=r'<a\\s+(?:[^>]*?\\s+)?href="([^"]*(?=index)[^"]*)"',
```

**Review by efriis — COMMENTED** [review]
Ah missed that this was embedded in a docstring sorry about that! I think this is needed to keep the code block accurate (the docs will render a single backslash, which means the string needs to be defined as an r-string)

**Comment by tibor-reiss** [comment]
> Ah missed that this was embedded in a docstring sorry about that! I think this is needed to keep the code block accurate (the docs will render a single backslash, which means the string needs to be defined as an r-string)

@efriis I am surprised that despite the double backslash you still need the `r` to mark it as raw string...

**Pull request merged**